### PR TITLE
Add `clean-header-search-field` feature

### LINKED ISSUE #4095: Hide duplicate query from header search field
Like https://github.com/sindresorhus/refined-github/issues/1253

Screen Shot 4

Not sure if this is a bug or if that feature was only added to global results pages.

Test URLs:

- https://github.com/sindresorhus/refined-github/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Abug
- https://github.com/pulls?q=is%3Apr+is%3Aopen+author%3Afregante+archived%3Afalse+sort%3Aupdated-desc (here it works)

### PR DESCRIPTION
**Pull request by cheap-glitch**
Closes #4095 

## Test URLs
* Search field should be cleared (**even when the page is reloaded**):
  * https://github.com/sindresorhus/refined-github/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Abug
  * https://github.com/pulls?q=is%3Apr+is%3Aopen+author%3Afregante+archived%3Afalse+sort%3Aupdated-desc
* Search field should NOT be cleared:
  * https://github.com/search?q=search
  * https://github.com/sindresorhus/refined-github/search?q=search

## Screenshot
clean-header-search-field

### CODE REVIEW

File: source/features/clean-header-search-field.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import onetime from 'onetime';
+import elementReady from 'element-ready';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+async function init(): Promise<void> {
+	(await elementReady<HTMLInputElement>('input.header-search-input'))!.value = '';
```
Comment (kidonng) [inline source/features/clean-header-search-field.tsx:8]: Is the leading `input` for the type inference? If I'm right `element-ready` doesn't support it yet, and you are already specifying `HTMLInputElement` here.

```suggestion
	(await elementReady<HTMLInputElement>('.header-search-input'))!.value = '';
```

File: readme.md [hunk 2]
```diff
@@ -215,6 +215,7 @@ Thanks for contributing! 🦋🙌
 - [](# "dim-bots") [Dims commits and PRs by bots to reduce noise.](https://user-images.githubusercontent.com/1402241/65263190-44c52b00-db36-11e9-9b33-d275d3c8479d.gif)
 - [](# "esc-to-cancel") [Adds a shortcut to cancel editing a conversation title: <kbd>esc</kbd>.](https://user-images.githubusercontent.com/35100156/98303086-d81d2200-1fbd-11eb-8529-70d48d889bcf.gif)
 - [](# "no-duplicate-list-update-time") [Hides the update time of conversations in lists when it matches the open/closed/merged time.](https://user-images.githubusercontent.com/1402241/111357166-ac3a3900-864e-11eb-884a-d6d6da88f7e2.png)
+- [](# "clean-header-search-field") [Clears duplicate queries in the header search field.](https://user-images.githubusercontent.com/46634000/114000463-5165bd00-985b-11eb-9389-8d0f145f5abe.png)
```
Comment (fregante) [inline readme.md:218]: The screenshot should be in higher resolution, probably have a before/after to show what’s happening, and appear in the GitHub shortcoming section

**Review by fregante — APPROVED** [review]

File: source/features/clean-header-search-field.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+	(await elementReady<HTMLInputElement>('.header-search-input'))!.value = '';
+}
+
+void features.add(__filebasename, {
+	include: [
+		pageDetect.isConversationList,
+		pageDetect.isGlobalConversationList
+	],
+	awaitDomReady: false,
+	init: onetime(init)
```
Comment (yakov116) [inline source/features/clean-header-search-field.tsx:17]: I did not blame this. But what is the reason for the one time?

Comment (fregante) [reply source/features/clean-header-search-field.tsx:17]: My guess is that the header is outside the ajaxed area, so it doesn't change after navigation. I just tested it and it works:

1. Visit https://github.com/sindresorhus/refined-github/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Abug
2. Click "795 Closed"
3. Go back

Field still empty

**Comment by cheap-glitch** [comment]
I think GitHub fixed it? I can't reproduce the bug anymore on Firefox with RGH disabled.

File: readme.md [hunk 4]
```diff
@@ -354,6 +354,7 @@ Thanks for contributing! 🦋🙌
 - [](# "same-page-definition-jump") [Avoids re-loading the page when jumping to function definition in the current file.](https://user-images.githubusercontent.com/16872793/90833649-7a5e2f80-e316-11ea-827d-a4e3ac8ced69.png)
 - [](# "linkify-full-profile-readme-title") [Linkifies the readme text on profile pages.](https://user-images.githubusercontent.com/16872793/90910173-ebe4bf00-e3a4-11ea-8fc5-aea3d1a2e5e5.png)
 - [](# "convert-pr-to-draft-improvements") [Moves the "Convert PR to Draft" button to the mergeability box and adds visual feedback to its confirm button.](https://user-images.githubusercontent.com/1402241/95644892-885f3f80-0a7f-11eb-8428-8e0fb0c8dfa5.gif)
+- [](# "clean-header-search-field") [Clears duplicate queries in the header search field.](https://user-images.githubusercontent.com/46634000/114152984-26dd3800-991f-11eb-808e-895a7e2545e3.png)
```
Comment (fregante) [inline readme.md:357]: Still doesn't do it 😅 makes me feel like I'm playing "Find the differences". I changed it to

Screen Shot

```suggestion
- [](# "clean-header-search-field") [Clears duplicate queries in the header search field.](https://user-images.githubusercontent.com/1402241/114177338-7c890300-9966-11eb-83a3-a711a76fae99.png)
```

**Pull request merged**